src/guide/host-functions/interacting-with-environment.rst

```diff
@@ -125,23 +125,50 @@ Below are several examples of how environment macro properties can be updated in
     # Define an host function called write_env_hostfn
     class write_env_hostfn(pyflamegpu.HostFunction):
       def run(self,FLAMEGPU):
-      # Retrieve the environment macro property foo of type float
-      foo = FLAMEGPU->environment.getMacroPropertyFloat("foo");
-      # Retrieve the environment macro property bar of type int array[3][3][3]
-      bar = FLAMEGPU.environment.getMacroPropertyInt("bar");
-      # Update some of the values
-      # foo = 12.0; is not allowed
-      foo.set(12.0);
-      foo[0] = 12.0; # This is the same as calling set()
-      bar[0][0][0]+= 1;
-      bar[0][1][0] = 5;
-      bar[0][0][2]+= 1; # Python does not allow the increment operator to be overridden
+          # Retrieve the environment macro property foo of type float
+          foo = FLAMEGPU->environment.getMacroPropertyFloat("foo");
+          # Retrieve the environment macro property bar of type int array[3][3][3]
+          bar = FLAMEGPU.environment.getMacroPropertyInt("bar");
+          # Update some of the values
+          # foo = 12.0; is not allowed
+          foo.set(12.0);
+          foo[0] = 12.0; # This is the same as calling set()
+          bar[0][0][0]+= 1;
+          bar[0][1][0] = 5;
+          bar[0][0][2]+= 1; # Python does not allow the increment operator to be overridden
       
 .. warning::
   Be careful when using :class:`HostMacroProperty<flamegpu::HostMacroProperty>` via the C++ API. When you retrieve an element e.g. ``bar[0][0][0]`` (from the example above), it is of type :class:`HostMacroProperty<flamegpu::HostMacroProperty>` not ``int``. Therefore you cannot pass it directly to functions which take generic arguments such as ``printf()``, as it will be interpreted incorrectly. You must either store it in a variable of the correct type which you instead pass, or explicitly cast it to the correct type when passing it e.g. ``(int)bar[0][0][0]`` or ``static_cast<int>(bar[0][0][0])``.
     
+Macro Property File Input/Output
+--------------------------------
+
+Environment macro properties are suitable for exceptionally large datasets. For this reason it may be necessary to initialise them from file. As such, the :class:`HostEnvironment<flamegpu::HostEnvironment>` provides methods for importing and exporting macro properties. Unlike model state export, these operate on a single property. The additional ``.bin`` (binary) file format is supported.
```

Comment:
`exceptionally large` feels a bit sensationalist, still limited by device memory which I wouldn't call exceptionally large. 

This is very nitpicky though, so feel free to ignore.